docs/autogen/templates/getting_started.rst.jinja

```diff
@@ -0,0 +1,726 @@
+:ref:`sec-gs-system-config-userspace-config` and if issues persist a solution
+might be found in the following subsections.
+
+No devices found
+~~~~~~~~~~~~~~~~
+
+When running ``xnvme enum`` then the output-listing is empty, there are no
+devices. When running with ``vfio-pci`` then this can occur when your devices
+are sharing iommu-group with other devices which are still bound to in-kernel
+drivers. This could be NICs, GPUs or other kinds of peripherals.
+
+The division of devices into groups is not something that can be easily
+switched, but you try to manually unbind the other devices in the iommu group
+from their kernel drivers.
+
+If that is not an option then you can try to re-organize your physical
+connectivity of deviecs, e.g. move devices around.
+
+Lastly you can try using ``uio_pci_generic`` instead, this can most easily be
+done by disabling iommu by adding the kernel option: ``iommu=off`` to the
+kernel command-line and rebooting.
+
+Memory Issues
+~~~~~~~~~~~~~
+
+If you see a message similar to the below while unbind devices::
+
+  Current user memlock limit: 16 MB
+
+  This is the maximum amount of memory you will be
+  able to use with DPDK and VFIO if run as current user.
+  To change this, please adjust limits.conf memlock limit for current user.
+
+  ## WARNING: memlock limit is less than 64MB
+  ## DPDK with VFIO may not be able to initialize if run as current user.
+
+Then go you should do as suggested, that is, adjust ``limits.conf``, see for an
+example of doing here :ref:`sec-gs-system-config-userspace-config`.
+
+Build Errors
+------------
+
+If you are getting errors while attempting to configure and build **xNVMe**
+then it is likely due to one of the following:
+
+* You are building in an **offline** environment and only have a shallow
+  source-archive or a git-repository without subprojects.
+
+The full source-archive is made available with each release and downloadble
+from the `GitHUB Release page <https://github.com/OpenMPDK/xNVMe/releases>`_
+release page. It contains the xNVMe source code along with all the third-party
+dependencies, namely: SPDK, liburing, libnvme, and fio.
+
+**missing dependencies / toolchain**
+
+* You are missing dependencies, see the :ref:`sec-building-toolchain` for
+  installing these on FreeBSD and a handful of different Linux Distributions
+
+The :ref:`sec-building-toolchain` section describes preferred ways of
+installing libraries and tools. For example, on Ubuntu 18.04 it is preferred to
+install ``meson`` via ``pip3`` since the version in the package registry is too
+old for SPDK, thus if installed via the package manager then you will
+experience build errors as the xNVMe build system starts building SPDK.
+
+Once you have the full source of xNVMe, third-party library dependencies, and
+setup the toolchain then run the following to ensure that the xNVMe repository
+is clean from any artifacts left behind by previous build failures::
+
+  make clobber
+
```

Comment:
If we want to suggest this, then we should probably include an disclaimer about `make clobber`.

Reply:
done.